src/components/MainTable/MainTable.tsx

```diff
@@ -190,60 +191,20 @@ const generateHeaders = (
   );
 };
 
-const generateRows = (
-  currentSortDirection: SortDirection,
-  currentSortKey: MainTableHeader["sortKey"],
-  emptyStateMsg: ReactNode,
-  expanding: Props["expanding"],
-  responsive: Props["responsive"],
-  headers: Props["headers"],
-  paginate: Props["paginate"],
-  rows: Props["rows"],
-  currentPage: number,
-  setCurrentPage: (page: number) => void,
-  sortable: Props["sortable"],
-  sortFunction: Props["sortFunction"]
-) => {
-  // If the table has no rows, return empty state message
-  if (Object.entries(rows).length === 0 && emptyStateMsg) {
-    return <caption>{emptyStateMsg}</caption>;
-  }
-  // Clone the rows so we can restore the original order.
-  const sortedRows = [...rows];
-  if (sortable && currentSortKey) {
-    if (!sortFunction) {
-      sortFunction = (a, b) => {
-        if (!a.sortData || !b.sortData) {
-          return 0;
-        }
-        if (a.sortData[currentSortKey] > b.sortData[currentSortKey]) {
-          return currentSortDirection === "ascending" ? 1 : -1;
-        } else if (a.sortData[currentSortKey] < b.sortData[currentSortKey]) {
-          return currentSortDirection === "ascending" ? -1 : 1;
-        }
-        return 0;
-      };
-    }
-    sortedRows.sort((a, b) =>
-      sortFunction(a, b, currentSortDirection, currentSortKey)
-    );
-  }
-  let slicedRows = sortedRows;
-  if (paginate) {
-    const startIndex = (currentPage - 1) * paginate;
-    if (startIndex > rows.length) {
-      // If the rows have changed e.g. when filtering and the user is on a page
-      // that no longer exists then send them to the start.
-      setCurrentPage(1);
-    }
-    slicedRows = sortedRows.slice(startIndex, startIndex + paginate);
-  }
-  const rowItems = slicedRows.map(
+const generateRows = ({
+  rows,
+  headers,
+  responsive,
+  expanding,
+}: Partial<Props> & {
```

Comment:
I don't think `Partial<Props>` is the right type here as this function requires a specific subset of the props.